app/javascript/bundles/components/NewItemModal.tsx

```diff
@@ -1,65 +1,96 @@
-                    <InputGroup.Text id="inputGroup-sizing-default">$</InputGroup.Text>
-                </InputGroup.Prepend>
-                <FormControl
-                    placeholder="e.g. 3.00"
-                    aria-label="price"
-                    />
-            </InputGroup>
-            </Col>
-            </Row>
-        </Modal.Body>
-        <Modal.Footer>
-            <Button onClick={this.props.onAdd} variant='primary'>Add item</Button>
-            <Button onClick={this.props.onClose} variant='secondary'>Close</Button>
-        </Modal.Footer>
-    </Modal>
+                </Modal.Header>
+                <Modal.Body>
+                    <Row>
+                        <Col>
+                            <h5>What is the item called?</h5>
+                            <InputGroup className="mb-3">
+                                <FormControl
+                                    placeholder="e.g. Bananas"
+                                    aria-label="name"
+                                    aria-describedby="inputGroup-sizing-default"
+                                    value={this.state.itemName}
+                                    onChange={this.handleItemNameChange}
+                                />
+                            </InputGroup>
+                        </Col>
+                    </Row>
+                    <Row>
+                        <Col>
+                            <h5>How much are you willing to pay for it?</h5>
+                            <InputGroup className='mb-3'>
+                                <InputGroup.Prepend>
+                                    <InputGroup.Text id="inputGroup-sizing-default">$</InputGroup.Text>
+                                </InputGroup.Prepend>
+                                <FormControl
+                                    placeholder="e.g. 3.00"
+                                    aria-label="price"
+                                    value={this.state.itemPrice}
+                                    onChange={this.handleItemPriceChange}
+                                />
+                            </InputGroup>
+                        </Col>
+                    </Row>
+                </Modal.Body>
+                <Modal.Footer>
+                    <Button onClick={(): void => this.props.onAdd(this.state.itemName, this.state.itemPrice)} variant='primary'>Add item</Button>
+                    <Button onClick={this.props.onClose} variant='secondary'>Close</Button>
+                </Modal.Footer>
+            </Modal>
+
+        );
+    }
 
-    );
-  }
+    handleItemNameChange(event: React.ChangeEvent<HTMLInputElement>): void {
+        this.setState({ itemName: event.target.value })
+    }
+    handleItemPriceChange(event: React.ChangeEvent<HTMLInputElement>): void {
+        const input = event.target.value
+        const num = Number(input)
+        if (!isNaN(num) && (num * 100 - Math.floor(num * 100) === 0)) {
```

Comment:
I'd like this logic abstracted into a function.

So maybe something called `isValidPrice(potentialPrice: string)`. You can do `string` and convert in the function or do `number`.